interview-questions.md

```diff
@@ -9,49 +9,148 @@ Then, research the question to expand on your answer. Even if you feel you have
+  }
+};
+
+const numbers = [1, 2, 3, 4, 5];
+const doubledNumbers = numbers.map(function(number) {
+  return this.doubleValue() * number;
+}, myObject);
+
+console.log(doubledNumbers); // Output: [20, 40, 60, 80, 100]
+```
+
+    In this example, numbers arrays calls on the method .map() and passes a callback function that multiplies each of the elements in the array by myObject.doubleValue(). Passing myObject as the second argument to the .map(), it is specifying that the value of "this" within the callback function should be "myObject", since "myObject" has a method called doubleValue that can be called within the callback function.
 
 3. What is the difference between map and filter?
 
 Your answer:
+    One of differences between .map() & .filter() is that while .map() will return an array with the same number of indexs, .filter() will not.
+    Another difference is that .map() can be used to access each element and manipulated the element's values such as multiplying each element of an array and return a new arraw with the new values.
 
-Researched answer:
+Researched answer: From OpenAI(ChatGPT4)
 
-1. What is iteration?
+    1. Purpose: 
+        .map() - is used to transform an array by applying a function to each element and returning a new array of the same length with the transfored elements.
+        .filter() - is used to create a new array with all elements that pass a certain test, based on a provided function
 
-Your answer:
+    2. Return Value:
+        .map() - will always returns a new array of the same length as the original array. 
+        .filter() - will always return a new array that may have less elelements than the original array, and that depends on the critirea the the filter function specifies
 
-Researched answer:
+    3. Function Signature
+        .map() - takes a callback function that is executed o each element of the array, and returns an array of the same length
+        .filter() - takes a callback function that is executed on each eleletn of the array, and returns an array containing only the lelemtns for which the coallback function returned "True"
 
-1. What is the difference between a function and a method?
+4. What is iteration?
 
 Your answer:
+    Iteration is the process of repeating a set of instruction over and over again until certian(s) condition(s) are met.
+
+Researched answer: From OpenAI(ChatGPT4)
+    Iteration refers to the process of performing repeatedly a set of operations or instructions on a sequence of elements or data. Iteration is a fundimental concept in computer programming that allows programmers to execute a block of code over and over for a number of specified times or until a condition is met. 
+    
+    In JavaScript, it is commonly performed using loops, such as the for loop, while loop, and do-while loop. With these loops a programmer can repeatedly execute a block of code while a certain condition is true, or until completing a specified number iterations.
+
```

Comment:
Great, love that you mentioned for loops. .map is another example of an iterative method.